lib/mix/tasks/oidcc.gen.provider_configuration_worker.ex

```diff
@@ -0,0 +1,140 @@
+defmodule Mix.Tasks.Oidcc.Gen.ProviderConfigurationWorker do
+  @example "mix oidcc.gen.provider_configuration_worker --name MyApp.OpenIDProvider"
+
+  @shortdoc "Generate an OpenID Connect provider configuration worker"
+  if !Code.ensure_loaded?(Igniter) do
+    @shortdoc "#{@shortdoc} | Install `igniter` to use"
+  end
+
+  @moduledoc """
+  #{@shortdoc}
+
+  Adds an `Oidcc.ProviderConfiguration.Worker` to your application and
+  configures it via the `runtime.exs` configuration file.
+
+  ## Example
+
+  ```bash
+  #{@example}
+  ```
+
+  ## Options
+
+  * `--name` or `-n` - The name of the provider configuration worker
+  * `--issuer` or `-i` - The issuer of the provider
+  """
+
+  if Code.ensure_loaded?(Igniter) do
+    use Igniter.Mix.Task
+
+    alias Igniter.Project.Application
+    alias Igniter.Project.Config
+
+    @impl Igniter.Mix.Task
+    def info(_argv, _composing_task) do
+      %Igniter.Mix.Task.Info{
+        # dependencies to add
+        adds_deps: [],
+        # dependencies to add and call their associated installers, if they exist
+        installs: [],
+        # An example invocation
+        example: @example,
+        # Accept additional arguments that are not in your schema
+        # Does not guarantee that, when composed, the only options you get are the ones you define
+        extra_args?: false,
+        # A list of environments that this should be installed in, only relevant if this is an installer.
+        only: nil,
+        # a list of positional arguments, i.e `[:file]`
+        positional: [],
+        # Other tasks your task composes using `Igniter.compose_task`, passing in the CLI argv
+        # This ensures your option schema includes options from nested tasks
+        composes: [],
+        # `OptionParser` schema
+        schema: [name: :string, issuer: :string],
+        # CLI aliases
+        aliases: [n: :name, i: :issuer]
+      }
+    end
+
+    @impl Igniter.Mix.Task
+    def igniter(igniter, argv) do
+      # extract positional arguments according to `positional` above
+      {_arguments, argv} = positional_args!(argv)
+      # extract options according to `schema` and `aliases` above
+      options = setup_options(argv, igniter)
+
+      igniter
+      |> configure_issuer(options)
+      |> add_application_worker(options)
+    end
+
+    defp setup_options(argv, igniter) do
+      argv
+      |> options!()
+      |> Keyword.update(
+        :name,
+        Igniter.Code.Module.module_name(igniter, "OpenIDProvider"),
+        &String.to_atom("Elixir.#{&1}")
```

Comment:
You can use `Igniter.Code.Module.parse/1` for this. It's just `String.split(".")` and `Module.concat`, but I think it has the same effect as what you're going for here?